filebeat/test_filebeat.py

```diff
@@ -0,0 +1,94 @@
+# (C) Datadog, Inc. 2010-2016
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+
+# stdlib
+from nose.plugins.attrib import attr
+from collections import namedtuple
+import os
+
+# 3p
+from mock import patch
+
+# project
+from tests.checks.common import AgentCheckTest, Fixtures
+
+
+mocked_file_stats = namedtuple('mocked_file_stats', ['st_size', 'st_ino', 'st_dev'])
+
+
+# allows mocking `os.stat` only for certain paths; for all others it will call
+# the actual function - needed as a number of test helpers do make calls to it
+def with_mocked_os_stat(mocked_paths_and_stats):
```

Comment:
I'm not sure I understand what you're suggesting, sorry? :)

The reason why this is somewhat complicated is that, as mentioned in the comment just above, a number of test helpers from `AgentCheckTest` _do_ need to stat real files, and so mocking `os.stat` altogether is not an option